src/Microsoft.AspNetCore.Razor.Tools/GenerateCommand.cs

```diff
@@ -0,0 +1,204 @@
+            ProjectDirectory = Option("-p", "project root directory", CommandOptionType.SingleValue);
+            OutputDirectory = Option("-o", "output directory", CommandOptionType.SingleValue);
+            TagHelperManifest = Option("-t", "tag helper manifest file", CommandOptionType.SingleValue);
+        }
+
+        public CommandArgument Sources { get; }
+
+        public CommandOption OutputDirectory { get; }
+
+        public CommandOption ProjectDirectory { get; }
+
+        public CommandOption TagHelperManifest { get; }
+
+        protected override Task<int> ExecuteCoreAsync()
+        {
+            var result = ExecuteCore(
+                projectDirectory: ProjectDirectory.Value() ?? Environment.CurrentDirectory,
+                outputDirectory: OutputDirectory.Value(),
+                tagHelperManifest: TagHelperManifest.Value(),
+                sources: Sources.Values.ToArray());
+
+            return Task.FromResult(result);
+        }
+
+        protected override bool ValidateArguments()
+        {
+            if (string.IsNullOrEmpty(OutputDirectory.Value()))
+            {
+                Error.WriteLine($"{OutputDirectory.ValueName} not specified.");
+                return false;
+            }
+
+            if (Sources.Values.Count == 0)
+            {
+                Error.WriteLine($"{Sources.Name} should have at least one value.");
+                return false;
+            }
+
+            return true;
+        }
+
+        private int ExecuteCore(string projectDirectory, string outputDirectory, string tagHelperManifest, string[] sources)
+        {
+            tagHelperManifest = Path.Combine(projectDirectory, tagHelperManifest);
+            outputDirectory = Path.Combine(projectDirectory, outputDirectory);
+
+            var tagHelpers = GetTagHelpers(tagHelperManifest);
+
+            var engine = RazorEngine.Create(b =>
+            {
+                RazorExtensions.Register(b);
+
+                b.Features.Add(new StaticTagHelperFeature() { TagHelpers = tagHelpers, });
+            });
+
+            var templateEngine = new MvcRazorTemplateEngine(engine, RazorProject.Create(projectDirectory));
+
+            var sourceItems = GetRazorFiles(projectDirectory, sources);
+            var results = GenerateCode(templateEngine, sourceItems);
+
+            var success = true;
+
+            foreach (var result in results)
+            {
+                if (result.CSharpDocument.Diagnostics.Count > 0)
+                {
+                    success = false;
+                    foreach (var error in result.CSharpDocument.Diagnostics)
+                    {
+                        Console.Error.WriteLine(error.ToString());
+                    }
+                }
+
+                var outputFilePath = Path.Combine(outputDirectory, Path.ChangeExtension(result.ViewFileInfo.ViewEnginePath.Substring(1), ".cs"));
```

Comment:
lol make these into some variables bro, this is debug hell waiting to happen 😉

Reply:
This was all copy paste. I am going to address all the comments in `GenerateCommand` as part of https://github.com/aspnet/Razor/issues/1953